layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/beagle-xm--Balena-environment.patch

```diff
@@ -0,0 +1,41 @@
+From 63f9d47f2cb04621ff1a5a7a1ae57bdb8bee9db0 Mon Sep 17 00:00:00 2001
+From: Sebastian Panceac <sebastian@balena.io>
+Date: Thu, 6 Jun 2019 13:55:47 +0200
+Subject: [PATCH] Integrate Beagle-XM with Balena
+
+Upstream-Status: Inappropriate [configuration]
+Signed-off-by: Sebastian Panceac <sebastian@balena.io>
+---
+ configs/omap3_beagle_defconfig | 2 +-
+ include/configs/omap3_beagle.h | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/configs/omap3_beagle_defconfig b/configs/omap3_beagle_defconfig
+index 4fb8aec57d..14d1428b09 100644
+--- a/configs/omap3_beagle_defconfig
++++ b/configs/omap3_beagle_defconfig
+@@ -5,7 +5,7 @@ CONFIG_TARGET_OMAP3_BEAGLE=y
+ CONFIG_SPL=y
+ CONFIG_DISTRO_DEFAULTS=y
+ CONFIG_NR_DRAM_BANKS=2
+-CONFIG_BOOTCOMMAND="run findfdt; run distro_bootcmd"
++CONFIG_BOOTCOMMAND="run findfdt; setenv resin_kernel_load_addr ${loadaddr}; run resin_set_kernel_root; setenv mmcdev ${resin_dev_index}; setenv bootpart ${resin_dev_index}:${resin_root_part}; run mmcboot"
+ CONFIG_SYS_CONSOLE_INFO_QUIET=y
+ CONFIG_DEFAULT_FDT_FILE="omap3-beagle.dtb"
+ CONFIG_VERSION_VARIABLE=y
+diff --git a/include/configs/omap3_beagle.h b/include/configs/omap3_beagle.h
+index 7698a90c4e..a1a33178db 100644
+--- a/include/configs/omap3_beagle.h
++++ b/include/configs/omap3_beagle.h
+@@ -151,7 +151,7 @@
+ 		"${mtdparts} " \
+ 		"${defargs} " \
+ 		"${optargs} " \
+-		"root=${mmcroot} " \
++		"${resin_kernel_root} " \
```

Comment:
you need also use os_cmdline feature from meta-balena. As an example:https://github.com/balena-os/balena-beaglebone/pull/189

Reply:
Thanks @floion. Fixed